src/SimpleReplay/extract/extractor/extractor.py

```diff
@@ -75,6 +81,38 @@ def get_extract(self, log_location, start_time, end_time):
             else:
                 return self.local_extractor.get_extract_locally(log_location, start_time, end_time)
 
+    def get_stored_procedures(self,start_time,end_time,username,stored_procedures_map,sql_json):
+        result_row ={}
+        parse_start_time = "'" + start_time + "'"
+        parse_end_time = "'" + end_time + "'"
+        for key,value in stored_procedures_map.items():
+            transaction_id = key
+            sys_query_history = f'select transaction_id, query_text \
+                from sys_query_history \
+                WHERE user_id > 1 \
+                AND transaction_id={transaction_id}\
+                AND start_time >= {parse_start_time} \
+                AND start_time <= {parse_end_time}\
+                ORDER BY 1;'
```

Comment:
1. Is this efficient? it seems like we could potentially end up hitting the source cluster with a lot of queries since it's in a while loop. Why can't we do this in batches of say 100 transaction ids?

2. It looks like you have a typo - it should be `AND end_time <= {parse_end_time}`.

Reply:
DO you mean adding a limit ?

Reply:
No. Something like this:

1. Split the `stored_procedures_map.items()` into groups of 100
2. Execute the query for each of the groups of 100 with `AND transaction_id IN (transaction_ids)`.
3. Group the results.

Reply:
done